lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -965,11 +969,25 @@ public function AbstractSchemaName()
      *
      * @throws QueryException if the name does not exist.
      */
-    private function validateAbstractSchemaName($schemaName)
+    private function validateAbstractSchemaName($schemaName) : void
     {
-        if (! (class_exists($schemaName, true) || interface_exists($schemaName, true))) {
-            $this->semanticalError("Class '$schemaName' is not defined.", $this->lexer->token);
+        // @TODO are these checks even needed?
```

Comment:
The check should probably be inversed: `getClassMetadata()` first, then `(class|interface)_exists()`. The reason being that `class_exists()` can trigger I/O, while `getClassMetadata()` usually doesn't.

This makes `SELECT new Foo(b.c) FROM Bar b` slower, as `getClassMetadata()` fails more often, but it makes other DQL queries faster.